### PR TITLE
Remove secret_controller and test; Use EnqueueRequestsFromMapFunc (#670)

### PR DESCRIPTION
**Pull request by keithchong**
#### Description:
- The reasons for reconciliation of the managed environment appears to be changed in the Jira description, in that it no longer checks if the namespace matches.  If this is no longer required, we simply remove the if block (I kept it in)
- The construction of the Namespace in the original code seems incorrect, in that it uses namespace.name instead of namespace.namespace.  I did not change this behaviour, but if you think it should remain at namespace.name, then we should leave it.

#### Link to JIRA Story (if applicable):
- See https://issues.redhat.com/browse/GITOPSRVCE-560

### CODE REVIEW

File: backend/controllers/managed-gitops/gitopsdeploymentmanagedenvironment_controller_test.go [hunk 1]
```diff
@@ -29,9 +31,11 @@ import (
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 )
 
 var _ = Describe("GitOpsDeploymentManagedEnvironment Controller Test", func() {
+	//ctx := context.Background()
```
Comment (jgwest) [inline backend/controllers/managed-gitops/gitopsdeploymentmanagedenvironment_controller_test.go:38]: Can remove this if no longer needed

**Review by jgwest — CHANGES_REQUESTED** [review]
Looks great, thanks @keithchong! A minor tweak then good to go.

File: backend/controllers/managed-gitops/gitopsdeploymentmanagedenvironment_controller_test.go (hunk 1 above)
Comment (keithchong) [reply backend/controllers/managed-gitops/gitopsdeploymentmanagedenvironment_controller_test.go:38]: Done

File: backend/controllers/managed-gitops/gitopsdeploymentmanagedenvironment_controller_test.go [hunk 2]
```diff
@@ -67,25 +73,142 @@ var _ = Describe("GitOpsDeploymentManagedEnvironment Controller Test", func() {
...
+
+					createManagedEnvTargetingSecret("testManagedEnv", secret, *namespace, k8sClient)
+
+					Expect(reconciler.findSecretsForManagedEnvironment(&secret)).To(BeEmpty())
+				})
+			})
+
+			When("Five GitOpsDeploymentManagedEnvironment reference a secret in the same namespace", func() {
+
+				It("should return two managed environments", func() {
```
[the remark this answers is not in the document]
Comment (keithchong) [reply backend/controllers/managed-gitops/gitopsdeploymentmanagedenvironment_controller_test.go:127]: Done

**Review by jgwest — APPROVED** [review]
LGTM, thanks @keithchong! And thanks for your patience in getting this approved :+1:

**Pull request merged**